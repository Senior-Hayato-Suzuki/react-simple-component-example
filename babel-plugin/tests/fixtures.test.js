const transformFileSync = require("babel-core").transformFileSync;
const fs = require("fs");

const fixtures = fs.readdirSync(__dirname + "/fixtures");


describe("compile fixtures to snapshots", () => {

    fixtures.forEach(fixtureFile => {

        test("compile: " + fixtureFile, () => {
            const actual = transformFileSync(__dirname + "/fixtures/" + fixtureFile, {
                plugins: [__dirname + "/../plugin.js"],
            });
            expect(actual.code).toMatchSnapshot();
        });

    });

});
