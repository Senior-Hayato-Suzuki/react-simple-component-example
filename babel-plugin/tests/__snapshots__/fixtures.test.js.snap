exports[`compile fixtures to snapshots compile: alternative-name.js 1`] = `
"import create from \"react-simple\";

const Foo = create(\"div\", {
    color: \"red\"
});
Foo.displayName = \"Foo\";"
`;

exports[`compile fixtures to snapshots compile: argument-shadow.js 1`] = `
"import simple from \"react-simple\";

function fn(simple) {
    var no = simple();
}"
`;

exports[`compile fixtures to snapshots compile: basic.js 1`] = `
"import simple from \"react-simple\";

const Foo = simple(\"div\", {
    color: \"red\"
});
Foo.displayName = \"Foo\";"
`;

exports[`compile fixtures to snapshots compile: export.js 1`] = `
"import simple from \"react-simple\";

export const Foo = simple(\"div\", {
    color: \"red\"
});
Foo.displayName = \"Foo\";"
`;

exports[`compile fixtures to snapshots compile: inside-function.js 1`] = `
"
import simple from \"react-simple\";

function fn() {
    const Foo = simple(\"div\", {
        color: \"red\"
    });
    Foo.displayName = \"Foo\";
}"
`;
